match/lib/match/storage/gitlab/client.rb

```diff
@@ -69,7 +69,23 @@ def upload_file(current_file, target_file)
               "name" => target_file
             )
 
-            execute_request(url, request)
+            response = execute_request(url, request)
+
+            log_upload_error(response, target_file) if response.code != "201"
+          end
+        end
+
+        def log_upload_error(response, target_file)
+          begin
+            response_body = JSON.parse(response.body)
+          rescue JSON::ParserError
+            response_body = response.body
+          end
+
+          if response_body["message"]["name"] == ["has already been taken"]
```

Comment:
Is `message` guaranteed to be in the JSON body? I think this would raise an error otherwise. Also looks like this could be a string if line 82 runs.

Reply:
Good call, this section is a bit clunky. I added an additional check to the conditional to ensure `message` is present. I also added more tests around that bit of code.